lldb/test/API/functionalities/scripted_platform/my_scripted_platform.py

```diff
@@ -6,20 +6,27 @@
 
 class MyScriptedPlatform(ScriptedPlatform):
     def __init__(self, exe_ctx, args):
-        self.processes = {}
-
-        proc = {}
-        proc["name"] = "a.out"
-        proc["arch"] = "arm64-apple-macosx"
-        proc["pid"] = 420
-        proc["parent"] = 42
-        proc["uid"] = 501
-        proc["gid"] = 20
-        self.processes[420] = proc
+        super.__init__(exe_ctx, args)
+
+        if args and args.GetType() == lldb.eStructuredDataTypeDictionary:
+            processes = args.GetValueForKey("processes")
+            for i in range(0, processes.GetSize()):
+                proc_info = processes.GetItemAtIndex(i)
+                proc = {}
+                proc["name"] = proc_info.GetValueForKey("name").GetStringValue(42)
+                proc["arch"] = proc_info.GetValueForKey("arch").GetStringValue(42)
+                proc["pid"] = proc_info.GetValueForKey("pid").GetIntegerValue()
+                proc["parent"] = proc_info.GetValueForKey("parent").GetIntegerValue()
+                proc["uid"] = proc_info.GetValueForKey("uid").GetIntegerValue()
+                proc["gid"] = proc_info.GetValueForKey("gid").GetIntegerValue()
```

Comment:
Can we document this dictionary format somewhere that users can see this and specify which ones are mandatory and which are optional?

Reply:
This is document on the `scripted_platform.py` base class and once this lands, the website will automatically be updated so users can see it. [This is what it should look like.](https://lldb.llvm.org/python_api/lldb.plugins.scripted_process.ScriptedProcess.html#lldb.plugins.scripted_process.ScriptedProcess.get_loaded_images)